### PR TITLE
[MM-32479] config/migrate_test: add cleanup

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
Add cleanup to the migrate test.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-32479

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: config/migrate_test.go [hunk 1]
```diff
@@ -73,6 +76,11 @@ func TestMigrate(t *testing.T) {
 			err = source.SetFile(file, []byte(filesData[i]))
 			require.NoError(t, err)
 		}
+
+		return func(store *config.Store) {
```
Comment (lieut-data) [inline config/migrate_test.go:80]: Should we leverage [`T.Cleanup`](https://golang.org/pkg/testing/#T.Cleanup) to simplify this further?

**Review by lieut-data — APPROVED** [review]

**Review by streamer45 — APPROVED** [review]
:+1:

File: config/migrate_test.go (hunk 1 above)
Comment (isacikgoz) [reply config/migrate_test.go:80]: Generally `t.Cleanup` comes handy but here I had to provide `config.Store` later on. If you had something different in your mind, can you elaborate a bit?

Comment (lieut-data) [reply config/migrate_test.go:80]: Ah, yeah, I see the difficulty here.

**Pull request merged**